### DISCUSSION TITLE
Migration From useAllDevices: true to Enumerated LVM Devices

### DISCUSSION THREAD

**Discussion by gchallen** [rook/rook#13357] [Q&A] [answered]
I maintain a small Rook Ceph cluster currently configured with `useAllDevices: true`. Each node has two small 16GB disks which have been picked up and are in use by the cluster.

At some point I realized that this means that there are two separate OSD pods running on every node, which seems like a waste of resources, particularly given that the underlying disks are so small. So I'd like to migrate the cluster to using LVMs instead to reduce the number of OSD pods by half.

My understanding is that Rook Ceph now supports LVMs, but they need to be specified directly in the cluster configuration and are not picked up by `useAllDevices: true`. That's fine, but my first question is how to migrate the cluster configuration to ensure that my existing storage devices continue to work during the migration. If I remove `useAllDevices: true`, does that only affect new device discovery? Will existing devices continue to work?

At that point I feel like the process should be straightforward:

1. Stop the operator to prevent new device discovery and OSD creation
1. Remove the OSDs from Ceph and wait for the cluster to heal
1. Delete the OSD pods on the current node
1. Merge the disks to create the new LVM
1. Update the cluster configuration to add the node and device to the storage map
1. Restart the operator and check that it discovers the new LVM disk
1. Repeat 15 more times

Does this sound about right? I'd appreciate any guidance from someone who has done something similar. I have live data on the cluster and, although it's backed up, I'd like to do this migration without downtime.

**Comment by travisn** [comment, marked as answer]
Some thoughts:
- When you remove the OSDs from ceph, see the [OSD removal guide](https://rook.io/docs/rook/latest/Storage-Configuration/Advanced/ceph-osd-mgmt/#remove-an-osd).
- `useAllDevices: true` only affects new OSD creation and upgrading OSDs. If you set that to false and then add new LVM settings, all the existing OSDs will be left running, but ignored by Rook for future reconciles.

**Reply by gchallen** [reply]
Thanks so much! Finished this yesterday and everything went very smoothly, despite a few mistypings here and there on my part. I appreciate your help.
